crates/e2e-test-utils/src/engine_api.rs

```diff
@@ -51,18 +54,40 @@ impl<E: EngineTypes> EngineApiTestContext<E> {
     ) -> eyre::Result<B256>
     where
         E::ExecutionPayloadEnvelopeV3: From<E::BuiltPayload> + PayloadEnvelopeExt,
+        E::ExecutionPayloadEnvelopeV4: From<E::BuiltPayload> + PayloadEnvelopeExt,
     {
-        // setup payload for submission
-        let envelope_v3: <E as EngineTypes>::ExecutionPayloadEnvelopeV3 = payload.into();
-
         // submit payload to engine api
-        let submission = EngineApiClient::<E>::new_payload_v3(
-            &self.engine_api_client,
-            envelope_v3.execution_payload(),
-            versioned_hashes,
-            payload_builder_attributes.parent_beacon_block_root().unwrap(),
-        )
-        .await?;
+        let submission = if self
+            .chain_spec
+            .is_prague_active_at_timestamp(payload_builder_attributes.timestamp())
+        {
+            let requests = payload
+                .executed_block()
+                .unwrap()
+                .execution_outcome()
+                .requests
+                .first()
+                .unwrap()
+                .clone();
```

Comment:
we should likely track some improvement on this, it's not nice and i think we're doing it in some other place too